packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1718,12 +1767,24 @@ export class ContainerRuntime
 			}
 			return provideEntryPoint(this);
 		});
+
+		// If we loaded from pending state, then we need to skip any ops that are already accounted in such
+		// saved state
+		this.skipSavedCompressorOps = pendingRuntimeState?.pendingIdCompressorState !== undefined;
 	}
 
 	/**
 	 * Initializes the state from the base snapshot this container runtime loaded from.
 	 */
 	private async initializeBaseState(): Promise<void> {
+		if (
+			this.idCompressorMode === "on" ||
+			(this.idCompressorMode === "delayed" && this.connected)
+		) {
+			assert(this.pendingIdCompressorOps.length === 0, "no pending ops");
```

Comment:
Why should there be no pending ops here?

Reply:
This is called from loadRuntime, we should not have yet processed any ops, and thus not accumulated any ops. Will clarify comment.

Reply:
Oh yeah, that makes sense. I confused this with ops in pending state which is loaded during container runtime load!